### PR TITLE
Android Instrumentation Tests Improvements

### PR DESCRIPTION
**Pull request by louwers**
I want to make the Android Instrumentation Tests pass on CI so we can rely on them for PRs.

In this PR I am pulling the logs from AWS Device Farm and log them on GitHub (for debugging in case a test fails).

Had to ignore quite a few tests because they are flaky.

`Style.getPredefinedStyle("Streets")` was failing unpredictably because the MapTiler styles weren't loaded yet. I replaced it with `Style.getDefaultStyles()[0].url` wherever possible. Maybe I will define the styles needed for the tests near the tests instead of relying on the pre-defined styles in the core (that we want to get rid of at some point).

I'll try to improve things further in another PR.

### CODE REVIEW

File: platform/android/MapLibreAndroidTestApp/src/androidTest/java/org/maplibre/android/offline/OfflineDownloadTest.kt [hunk 1]
```diff
@@ -16,6 +17,7 @@ import java.util.concurrent.TimeoutException
 /**
  * Integration test that validates downloading an offline region from a point geometry at zoomlevel 17
  */
+@Ignore("https://github.com/maplibre/maplibre-native/issues/2318")
```
Comment (cgalvan) [inline platform/android/MapLibreAndroidTestApp/src/androidTest/java/org/maplibre/android/offline/OfflineDownloadTest.kt:20]: Thanks for creating these issues so we don't forget to come back and look at these flaky tests :)

File: scripts/aws-device-farm/s3-client.mjs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { S3Client } from "@aws-sdk/client-s3";
+
+export function getS3Client() {
+  return new S3Client({
+    region: "eu-central-1",
```
Comment (cgalvan) [inline scripts/aws-device-farm/s3-client.mjs:5]: Just wanted to call out that the device farm region is `us-west-2`. Should we have them on the same region, or do we want to test with them in different regions for more coverage?

**Review by cgalvan — APPROVED** [review]

File: scripts/aws-device-farm/s3-client.mjs (hunk 2 above)
Comment (louwers) [reply scripts/aws-device-farm/s3-client.mjs:5]: Our S3 bucket is in eu-central-1.

Comment (louwers) [reply scripts/aws-device-farm/s3-client.mjs:5]: Yeah the AWS Device Farm client uses `us-west-2`. 👍

**Pull request merged**